receiver/k8slogreceiver/config.go

```diff
@@ -0,0 +1,219 @@
+	Regex string `mapstructure:"regex"`
+}
+
+func (c Config) Validate() error {
+	return c.Discovery.Validate()
+}
+
+// SourceConfig allows specifying how to discover containers to collect logs from.
+type SourceConfig struct {
+	// Mode represents the mode of the k8slog receiver.
+	// Valid values are:
+	// - "daemonset-stdout": (default) otel is deployed as a daemonset and collects logs from stdout of containers.
+	//
+	// Will be supported in the future:
+	// - "daemonset-file": otel is deployed as a daemonset and collects logs from files inside containers.
+	// - "sidecar": otel is deployed as a sidecar and collects logs from files.
+	Mode string `mapstructure:"mode"`
+
+	// NodeFromEnv represents the environment variable which contains the node name.
+	NodeFromEnv string `mapstructure:"node_from_env"`
+
+	// HostRoot represents the path which is used to mount the host's root filesystem.
+	HostRoot string `mapstructure:"host_root"`
+
+	// K8sAPI represents the configuration for the k8s API.
+	K8sAPI k8sconfig.APIConfig `mapstructure:"k8s_api"`
+
+	// RuntimeAPIs represents the configuration for the runtime APIs.
+	RuntimeAPIs []RuntimeAPIConfig `mapstructure:"runtime_apis"`
+
+	Filter []FilterConfig `mapstructure:"filter"`
+}
+
+func (c SourceConfig) Validate() error {
+	var err error
+	if c.Mode != ModeDaemonSetStdout {
+		return fmt.Errorf("invalid mode %q", c.Mode)
+	}
+	if c.HostRoot == "" {
+		err = multierr.Append(err, fmt.Errorf("host_root must be specified when mode is %q", c.Mode))
+	}
+	err = multierr.Append(err, c.K8sAPI.Validate())
+	for _, r := range c.RuntimeAPIs {
+		err = multierr.Append(err, r.Validate())
+	}
+	return err
+}
+
+// FilterConfig allows specifying how to filter containers to collect logs from.
+// By default, all containers are collected from.
+type FilterConfig struct {
+	// Annotations represents the rules to filter containers based on pod annotations.
+	Annotations []MapFilterConfig `mapstructure:"annotations"`
+
+	// Labels represents the rules to filter containers based on pod labels.
+	Labels []MapFilterConfig `mapstructure:"labels"`
+
+	// Env represents the rules to filter containers based on pod environment variables.
+	Env []MapFilterConfig `mapstructure:"env"`
+
+	// Namespaces represents the rules to filter containers based on pod namespaces.
+	Namespaces []ValueFilterConfig `mapstructure:"namespaces"`
+
+	// Containers represents the rules to filter containers based on container names.
+	Containers []ValueFilterConfig `mapstructure:"containers"`
+
+	// Pods represents the rules to filter containers based on pod names.
+	Pods []ValueFilterConfig `mapstructure:"pods"`
+
+	// UIDs represents the rules to filter containers based on pod UIDs.
+	UIDs []ValueFilterConfig `mapstructure:"uids"`
+
+	// FilterStatements represents the ottl statements to filter containers.
+	// TODO: define the context.
+	FilterStatements []string `mapstructure:"filter_statements"`
```

Comment:
Since we don't have a k8s context in OTTL for this yet we can start without it. Lets remove it for now and add it back once we could implement it.

Reply:
This has been removed